### ISSUE TITLE
Support defining a default alert config in the provider config

### ISSUE THREAD

**Issue by dchidell** [TwinProduction/gatus#91]
The current alerting configuration allows for some flexible configuration, however it's often desirable to define the same alerting standard for multiple services.

I'd propose something like the following:

```
alerting:
  pagerduty: 
    integration-key: "********************************"
    templates:
      - name: twinnation_pager
        enabled: true
        failure-threshold: 3
        success-threshold: 5
        send-on-resolved: true
        description: "healthcheck failed 3 times in a row"

services:
  - name: twinnation_service_a
    url: "https://twinnation.org/health/a"
    interval: 30s
    alerts:
      - template: twinnation_pager
    conditions:
      - "[STATUS] == 200"
      - "[BODY].status == UP"
      - "[RESPONSE_TIME] < 300"

  - name: twinnation_service_b
    url: "https://twinnation.org/health/b"
    interval: 30s
    alerts:
      - template: twinnation_pager
    conditions:
      - "[STATUS] == 200"
      - "[BODY].status == UP"
      - "[RESPONSE_TIME] < 300"

**Comment by TwinProduction** [comment]
I feel like having the option to specify both an alert configuration template in the provider configuration and have the ability to specify it directly on the service is a little too confusing.

Instead, I believe that having a "default" alerting configuration at the provider level would be a great idea. 

The only parameter that would be required at the service level would be the type:


```yaml
alerting:
  pagerduty: 
    integration-key: "********************************"
    default:
      enabled: true
      failure-threshold: 3
      success-threshold: 5
      send-on-resolved: true
      description: "healthcheck failed 3 times in a row"

services:
  - name: example-service
    url: "https://twinnation.org/health"
    alerts:
      - type: pagerduty
    conditions:
      - "[STATUS] == 200"
```

Of course, it would also be possible to override the default configuration:
```yaml
services:
  - name: example-service
    url: "https://twinnation.org/health"
    alerts:
      - type: pagerduty
        description: "everything is on fire" 
    conditions:
      - "[STATUS] == 200"
```

What do you think?

**Comment by dchidell** [comment]
Your proposal fits my use case perfectly.

However my only concern around just having a type is if a user wishes to have multiple different configurations for different types / priorities / flavours of service. Overriding the configuration seems like a more than reasonable solution for this. It may add more config but it does keep things simple.

**Comment by TwinProduction** [comment]
Just merged the changes for this issue, should be available through the `latest` tag shortly.

@dchidell Would you mind giving it a try and letting me know if all works as expected?

**Comment by TwinProduction** [comment]
Released in [v2.6.0](https://github.com/TwinProduction/gatus/releases/tag/v2.6.0)
